File: src/code-samples/langsmith/smithdb-migration/runs-retrieve-basic-after.go
// :snippet-start: runs-retrieve-basic-after-go
// :codegroup-tab: After
package main

import (
	"context"
	"fmt"
	"time"

	"github.com/langchain-ai/langsmith-go"
)

// :remove-start:
func main() {
// :remove-end:
ctx := context.Background()
client := langsmith.NewClient()

runID := "<run-id>"
startTime := time.Date(2026, 6, 1, 12, 0, 0, 0, time.UTC)
projectID := "<project-id>"
// :remove-start:
sessions, err := client.Sessions.List(ctx, langsmith.SessionListParams{
	Name:  langsmith.F("default"),
	Limit: langsmith.F(int64(1)),
})
if err != nil {
	panic(err.Error())
}
projectID = sessions.Items[0].ID
found, err := client.Runs.QueryV2(ctx, langsmith.RunQueryV2Params{
	ProjectIDs: langsmith.F([]string{projectID}),
	Selects: langsmith.F([]langsmith.RunQueryV2ParamsSelect{
		langsmith.RunQueryV2ParamsSelectID,
		langsmith.RunQueryV2ParamsSelectStartTime,
	}),
	PageSize: langsmith.F(int64(1)),
})
if err != nil {
	panic(err.Error())
}
runID = found.Items[0].ID
startTime = found.Items[0].StartTime
// :remove-end:
run, err := client.Runs.GetV2(ctx, runID, langsmith.RunGetV2Params{
	ProjectID: langsmith.F(projectID),
	StartTime: langsmith.F(startTime),
	Selects: langsmith.F([]langsmith.RunGetV2ParamsSelect{
		langsmith.RunGetV2ParamsSelectName,
		langsmith.RunGetV2ParamsSelectStatus,
		langsmith.RunGetV2ParamsSelectTotalTokens,
	}),
})
// :remove-start:
if err != nil {
	panic(err.Error())
}
// :remove-end:
fmt.Println(run.Name, run.Status, run.TotalTokens)
// :remove-start:
}

// :remove-end:
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/runs-retrieve-basic-after.ts
import { Client } from "langsmith";

async function findRun(projectId: string) {
  const client = new Client();
  for await (const run of client.runs.query({ project_ids: [projectId], selects: ["ID", "START_TIME"] })) {
    return run;
  }
  return null;
}

async function getProjectId() {
  const client = new Client();
  const page = await client.projects.list({ name: "default", limit: 1 });
  const projects = page.getPaginatedItems();
  return projects[0]?.id;
}

// :snippet-start: runs-retrieve-basic-after-js
// :codegroup-tab: After
import { Client } from "langsmith";

const client = new Client();
let runId = "<run-id>";
let startTime = "2026-06-01T12:00:00Z";
let projectId = "<project-id>";
// :remove-start:
projectId = await getProjectId();
const run = await findRun(projectId);
runId = run.id;
startTime = run.start_time;
// :remove-end:
const retrievedRun = await client.runs.retrieve(runId, {
  project_id: projectId,
  start_time: startTime,
  selects: ["NAME", "STATUS", "TOTAL_TOKENS"],
});
console.log(retrievedRun.name, retrievedRun.status, retrievedRun.total_tokens);
// :snippet-end:
